app/services/api_test_generator.py

```
@@ -0,0 +1,358 @@
+        except Exception as e:
+            logger.error(f"Unexpected error during test generation: {e}", exc_info=True)
+            raise LoadTestError(
+                f"Test generation failed unexpectedly: {e}",
+                "GENERATION_UNEXPECTED_ERROR",
+                {
+                    "swagger_url": swagger_url,
+                    "operation_id": operation_id,
+                    "error": str(e),
+                    "error_type": type(e).__name__
+                }
+            )
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)